codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/RubySymbolProvider.java

```diff
@@ -318,14 +318,14 @@ public Symbol serviceShape(ServiceShape shape) {
     @Override
     public Symbol structureShape(StructureShape shape) {
         String name = getDefaultShapeName(shape, "Struct__");
-        return createSymbolBuilder(shape, name, name, name, moduleName)
+        return createSymbolBuilder(shape, name, "Types::" + name, name, moduleName)
                 .definitionFile(DEFAULT_DEFINITION_FILE).build();
     }
 
     @Override
     public Symbol unionShape(UnionShape shape) {
         String name = getDefaultShapeName(shape, "Union__");
-        return createSymbolBuilder(shape, name, name, name, moduleName)
+        return createSymbolBuilder(shape, name, "Types::" + name, name, moduleName)
```

Comment:
Is this correct? Some times we may not want to print out a full qualified name. I would assume it's relative to where it's being written.

Reply:
Yeah - I was debating this.  The `rbsType` property doesn't have the same namespace sensitive ability as the actual symbol name.  I think `Types::<name>` will always be correct (it just may be more information than necessary if you use it within the types module)